File: archives/2017-2018/lectures/6/server/portfolio.go
package main

type Portfolio struct {
	Id           int       `json:"id,omitempty"`
	Name         string    `json:"name,omitempty"`
	LastModified int64 `json:"lastModified,omitempty"`
	Symbols      []Symbol `json:"symbols,omitempty"`
}

type Portfolios []Portfolio

File: lectures/7/flutter_jwt.old/lib/portfolio.dart

class Portfolio {
  final int id;
  final String name;
  final int lastModified;

  Portfolio({required this.id, required this.name, required this.lastModified});

  factory Portfolio.fromJson(Map<String, dynamic> json) {
    return Portfolio(
      id: json['id'],
      name: json['name'],
      lastModified: json['lastModified'],
    );
  }

  static List<Portfolio> fromJsonList(List<dynamic> json) {
    List<Portfolio> portfolios = [];
    json.forEach((item) => portfolios.add(Portfolio.fromJson(item)));
    return portfolios;
  }

  @override
  String toString() {
    return 'Portfolio{id: $id, name: $name}';
  }

}